File: FC.App/Scripts/Modules/Tickets/Models/Ticket.ts
module FC.Shared.Models {
    export class Ticket {
        
        public TicketID :string;
        public AuthorID :string;
        public Price : number;
        public Author :FC.Shared.Models.ApplicationUser;
        public TicketName :string;
        public TicketDescription :string;
        public IsAvailable :boolean;
        public IsEarlyBird: boolean;
        public IsVipTicket: boolean;
        public IsDiscount: boolean;
        public IsCombiDeal: boolean;
        public IsAllinclusive: boolean;
        public ExternalTicketURL :string;
        public InternalURL :string;
        public Created :string;
        public Modified :string;
        public ArchiveDate :string;
        public Deleted: boolean;
        public CurrencyBase :string;
    }
}

File: FC.Shared/Entities/Ticket.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using FC.Shared.Enum;
using FC.Interfaces.Data;

namespace FC.Shared.Entities
{
    public class Ticket: BaseModel
    {
        public Ticket()
        {
            this.Created = DateTime.Now;
            this.Modified = null;
            this.IsAllinclusive = false;
            this.IsCombiDeal = false;
            this.IsDiscount = false;
            this.CurrencyBase = FC.Shared.Enum.CurrencyBase.EUR;
        }
        
        [Key]
        public Guid? TicketID { get; set; }

        [Index]
        public Guid? AuthorID { get; set; }

        public double Price { get; set; }

        [ForeignKey("AuthorID")]
        public ApplicationUser Author { get; set; }
        

        public string TicketDescription { get; set; }
        
        public bool IsAvailable { get; set; }
        public bool IsEarlyBird { get; set; }
        public bool IsVipTicket { get; set; }
        public bool IsDiscount { get; set; }
        public bool IsCombiDeal { get; set; }
        public bool IsAllinclusive { get; set; }
        public bool ExternalTicketURL { get; set; }
        public bool InternalURL { get; set; }
        
        public string CurrencyBase { get; set; }
    }
}
